modules/gapi/include/opencv2/gapi/garray.hpp

```diff
@@ -336,17 +338,21 @@ namespace detail
 template<typename T> class GArray
 {
 public:
+    // Host type (or Flat type) - the type this GArray is actually
+    // specified to.
+    using HT = typename detail::flatten_g<typename std::decay<T>::type>::type;
```

Comment:
is it correct to move `HT` from `private` to `public`? Were there any reasons to put it into `private` originally?

Reply:
We need HT in the public constructor. Put private zone under public zone - bad behavior.

Reply:
I assume you meant "put private above"

Yeah, agree that swapping them isn't a good idea

Anyway, it doesn't seem to cause any severe problems..

OK